reference-controllers/zookeeper-controller/src/main.rs

```diff
@@ -217,48 +217,51 @@ async fn reconcile_stateful_set(zk: &ZookeeperCluster, client: Client) -> Result
             ..old_sts
         };
         // Scale up or no change
-        sts_api
-            .replace(sts_name, &PostParams::default(), &updated_sts)
-            .await
-            .map_err(Error::ReconcileStatefulSetFailed)?;
-
         if updated_sts.spec.as_ref().unwrap().replicas.unwrap()
             < old_sts.spec.as_ref().unwrap().replicas.unwrap()
```

Comment:
It seems to me we don't need to check whether it is a scale down now? because updating the CLUSTER_SIZE does not harm to scale up.